src/main/python/fine_tuning/run_batch.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+"""
+Anserini: A toolkit for reproducible information retrieval research built on Lucene
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import os
+import subprocess
+import argparse
+from multiprocessing import Pool
+import json
+import logging
+
+import yaml
+
+from search import Search
+from evaluation import Evaluation
+from effectiveness import Effectiveness
+
+logger = logging.getLogger('fine_tuning')
+logger.setLevel(logging.INFO)
+# create console handler with a higher log level
+ch = logging.StreamHandler()
+ch.setLevel(logging.INFO)
+formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+ch.setFormatter(formatter)
+# add the handlers to the logger
+logger.addHandler(ch)
+
+parallelism=1
+def batch_everything(all_params, func):
+    if len(all_params) == 0:
+        return
+    p = Pool(parallelism)
+    p.map(func, all_params)
+
+def isclose(a, b, rel_tol=1e-09, abs_tol=0.0):
```

Comment:
Since there is no "expected", we do not need `isclose` function as well?

Reply:
Let's keep it for now? We probably will use it in the future.